lib/src/analyzer/api/analyzer.dart

```diff
@@ -0,0 +1,74 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart';
+
+import '../../../flutter_code_editor.dart';
+import '../../job_runner/job_runner.dart';
+
+/// Service for analzying the code inside CodeField.
+///
+/// Uses streams to connect incoming new code and outgoing list of issues.
+/// Retrieves code from `CodeController` every second
+/// and adds to the stream for analysis.
+///
+/// Inherit and implement [analyze] method to use in [CodeController].
+abstract class Analyzer {
```

Comment:
This class should provide a method `Future<List<Issue>> analyze(Code)`. So no stream is required.

Reply:
Or rather `Future<AnalysisResult>`